### DISCUSSION TITLE
Is there a way to set workspace wise priority of Linters?

### DISCUSSION THREAD

**Discussion by V4G4X** [mason-org/mason.nvim#1431] [Q&A]
This case might be rare, or maybe you use YAMLs in different contexts too.  

So I have the following installed:
 - yaml-language-server (LSP)
 - cfn-lint (Linter)
 - yamlfix (Formatter)

Now note that I have 2 workspaces:
 - With CloudFormation templates (hence `cfn-lint`)
 - With OpenAPI specs

As you can guess, what I need is that when using the CloudFormation workspace, the cfn linter should be active over the default yaml linter. This is because the yaml linter can't understand the cfn related keywords and functions.  
All the other features of the yaml LSP should still be available.

And in the OpenAPI spec workspace, the yaml LSP should work as normal. Or maybe I can find an OpenAPI specific linter and that can take priority.

Is this possible?

P.S Also could there be any reason why yaml formatters like `yamlfmt` and `yamlfix` don't work and return `[LSP] Format Request failed, no matching language servers` even when `:LspInfo` shows "1 client(s) attached to this buffer".

**Comment by williamboman** [comment]
Hello! This depends on what tools you use to integrate the tools inside Neovim, and is unrelated to mason.nvim (which only installs these).
